src/ILCompiler.ReadyToRun/src/ObjectWriter/SectionBuilder.cs

```diff
@@ -923,24 +923,68 @@ private static void WriteDirectoryEntry(DirectoryEntry directoryEntry, BlobBuild
     /// </summary>
     public static class SectionBuilderExtensions
     {
+        public static void PrepareSections(
+            this SectionBuilder builder,
+            PEReader peReader,
+            Func<string, ISymbolNode> sectionStartNodeLookup,
+            HashSet<string> customSections,
+            out int textSectionIndex,
```

Comment:
Nit: This would look cleaner without the out params. Could we just ask `SectionBuilder` for the indexes after setting up the sections?

Reply:
Fixed in 2nd commit. The new concept of SectionBuilder encapsulation in R2RPEBuilder let me make ObjectWriter completely oblivious of the integral section indices.